crates/rpc/rpc-eth-api/src/helpers/transaction.rs

```diff
@@ -436,6 +436,23 @@ pub trait EthTransactions: LoadTransaction<Provider: BlockReaderIdExt> {
         }
     }
 
+    /// Fills the defaults on a given unsigned transaction.
+    fn fill_transaction(
+        &self,
+        request: RpcTxReq<Self::NetworkTypes>,
+    ) -> impl Future<Output = Result<FillTransactionRes<RpcTxReq<Self::NetworkTypes>>, Self::Error>> + Send
+    where
+        Self: EthApiSpec,
+    {
+        async move {
+            // TODO: implement transaction filling logic
+            let _ = request;
+            Err(Self::Error::from_eth_err(EthApiError::Unsupported(
+                "eth_fillTransaction not yet implemented",
+            )))
```

Comment:
here's we'd need to derive the txtype and fill missing fields (mostly gas).

it looks like the reference impl doesnt do any fancy gaslimit estimation or use suggested fees, but I feel we should do this here?